File: memory-domain/src/main/java/com/memory/domain/todo/repository/TodoRepositoryCustom.java
package com.memory.domain.todo.repository;

import com.memory.domain.member.Member;
import com.memory.domain.todo.Todo;

import java.time.LocalDateTime;
import java.util.List;

public interface TodoRepositoryCustom {

    List<Todo> findByMemberAndDueDateBetween(Member member, LocalDateTime startDateTime, LocalDateTime endDateTime);

}

File: memory-domain/src/main/kotlin/com/memory/domain/todo/repository/TodoRepositoryCustom.kt
package com.memory.domain.todo.repository

import com.memory.domain.member.Member
import com.memory.domain.routine.Routine
import com.memory.domain.todo.Todo
import java.time.LocalDateTime

interface TodoRepositoryCustom {

    fun findByMemberAndDueDateBetween(member: Member?, startDateTime: LocalDateTime?, endDateTime: LocalDateTime?): List<Todo>

    fun existsByMemberAndRoutineAndDueDate(member: Member?, routine: Routine?, dueDate: LocalDateTime?): Boolean
}
